### PR TITLE
COP-3190 & COP-2758: Add mobile number validation and mark vessel fields as optional

### PR DESCRIPTION
**Pull request by ajames14**
This branch marks the fields on the create a vessel page as optional and adds validation to the mobile number when a user registers (allows numbers and whitespaces only).

To test:
When signed out, attempt to register a user with a phone number that contains anything other than a digit or whitespace. Submit and you will see an error message.
Navigate to the create a vessel page (http://localhost:8080/vessels/save-vessel?source=vessels) - you will see optional on the relevant field labels.

## Developer Checklist

\* Required

- [x] Up-to-date with master branch? \*

- [x] Does it have tests?

- [ ] Pull request URL linked to JIRA ticket? \*

- [ ] All reviewer comments resolved/answered? \*

Ensure you delete the branch once the Pull Request is merged.

### CODE REVIEW

File: src/components/Forms/validationRules.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 export const VALID_EMAIL_REGEX = /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i;
 export const COMPLEX_PASSWORD_REGEX = /(\d+)|([a-z]+)|([A-Z]+)|([!@$%^&\\(){}[\]:;<>,*.?/~_+-=|]+)+/g;
+export const VALID_MOBILE_REGEX = /^\d([ 0-9]{3,20})$/i;
```
Comment (rafenden) [inline src/components/Forms/validationRules.js:3]: I'm not sure if that’s going to be enough, what if the user enters country code, e.g. +44123456778 or dashes?

File: src/components/User/UserRegisterValidation.jsx [hunk 2]
```diff
@@ -22,10 +23,11 @@ const UserRegisterValidation = (dataToValidate) => {
   }
 
   // Mobile Number field must be valid
-  if (!parseInt(dataToValidate.mobileNumber, 10)) {
+  if (!(VALID_MOBILE_REGEX.test(dataToValidate.mobileNumber))) {
     fieldsErroring.mobileNumber = 'You must enter a valid phone number';
```
Comment (rafenden) [inline src/components/User/UserRegisterValidation.jsx:27]: IMO, we should include also a hint under the text field containing an example of a valid phone number, as the validation might be strict in some cases, i.e. (doesn't allow dashes).

File: src/components/Forms/validationRules.js (hunk 1 above)
Comment (ajames14) [reply src/components/Forms/validationRules.js]: Updated to allow leading +, but no dashes.

File: src/components/User/UserRegisterValidation.jsx (hunk 2 above)
Comment (ajames14) [reply src/components/User/UserRegisterValidation.jsx]: Spoke to Christina and have included an explanatory error message and added a hint

**Comment by Bothees** [comment]
LGTM 👍

**Review by Bothees — APPROVED** [review]
👍

**Pull request merged**